platform/linux-generic/pktio/dpdk.c

```diff
@@ -1684,8 +1685,15 @@ static int dpdk_open(odp_pktio_t id ODP_UNUSED,
 		return -1;
 	pool_entry = pool_entry_from_hdl(pool);
 
-	if (!dpdk_netdev_is_valid(netdev)) {
-		ODP_ERR("Invalid dpdk netdev: %s\n", netdev);
+	/* Init pktio entry */
+	memset(pkt_dpdk, 0, sizeof(*pkt_dpdk));
+
+	if (!rte_eth_dev_get_port_by_name(netdev, &port_id))
+		pkt_dpdk->port_id = port_id;
+	else if (dpdk_netdev_is_valid(netdev))
+		pkt_dpdk->port_id = atoi(netdev);
+	else {
+		ODP_ERR("Invalid DPDK interface name!: %s\n", netdev);
```

Comment:
You can still remove the exclamation mark but otherwise the PR is looking good.

Reply:
Changed the log message and added reviewed-by tag